File: engine/card.py
from .state import BattleState
from .symptom import trigger_suppress

CARD_DATA = None

def load_cards(path):
    global CARD_DATA
    import json
    with open(path) as f:
        CARD_DATA = {c["id"]: c for c in json.load(f)}

def play_card(st: BattleState, card_id: str):
    c = CARD_DATA[card_id]
    e = c["effect"]
    ctype = c["type"]

    st.msg(f"  > {card_id} 사용")

    if ctype == "treatment":
        dmg = e.get("damage", 0)
        # 반감 효과 체크 (고열경련)
        from .symptom import SYMPTOM_DATA
        for s in st.symptoms:
            if s.suppressed == 0 and SYMPTOM_DATA.get(s.name, {}).get("treatment_half"):
                dmg //= 2
                st.msg(f"    [{s.name}] 치료 효과 반감")
                break
        # 버프/디버프 적용
        dmg = max(0, dmg + st.treatment_buff - st.treatment_debuff)
        st.treatment_buff = 0  # 소모성 버프

        if "suppress" in e:
            _do_suppress(st, e)
        elif dmg:
            st.disease_hp -= dmg
            st.msg(f"    본체 -{dmg} HP → {st.disease_hp}")

    elif ctype == "stabilize":
        def_val = e.get("defense", 0)
        heal = e.get("heal", 0)
        st.defense_total += def_val
        if heal:
            st.patient_hp = min(st.max_patient_hp, st.patient_hp + heal)
            st.msg(f"    환자 +{heal} HP → {st.patient_hp}")
        if def_val:
            st.msg(f"    방어 +{def_val} (누적 {st.defense_total})")
        if e.get("reveal"):
            _reveal(st, e["reveal"])
        if e.get("reveal_all"):
            st.msg("    모든 증상 의도 확인")

    elif ctype == "support":
        if "buff_next_treatment" in e:
            st.treatment_buff += e["buff_next_treatment"]
            st.msg(f"    다음 치료 카드 +{e['buff_next_treatment']}")
        if "cost_reduce_next" in e:
            st.cost_reduce_next += e["cost_reduce_next"]
            st.msg(f"    다음 카드 코스트 -{e['cost_reduce_next']}")
        if "draw" in e:
            st.draw_cards(e["draw"])
            st.msg(f"    {e['draw']}장 드로우")
        if "buff_treatment_this_turn" in e:
            st.treatment_buff += e["buff_treatment_this_turn"]
            st.msg(f"    이번 턴 치료 카드 +{e['buff_treatment_this_turn']}")
        if e.get("reveal_all"):
            st.msg("    모든 증상 의도 확인")

def _do_suppress(st: BattleState, effect):
    from .symptom import SYMPTOM_DATA
    target = effect.get("suppress")
    turns = effect.get("turns", 2)

    active = [s for s in st.symptoms if s.suppressed == 0]
    if not active:
        st.msg("    억제할 활성 증상 없음")
        return

    if target == "any":
        # CLI: 플레이어 선택
        print("    억제할 증상 선택:")
        for i, s in enumerate(active):
            print(f"      {i}. {s.name}")
        while True:
            try:
                idx = int(input("      번호: "))
                chosen = active[idx]
                break
            except (ValueError, IndexError):
                print("      다시 입력")
    else:
        chosen = next((s for s in active if s.name == target), None)
        if not chosen:
            st.msg(f"    {target} 활성 아님")
            return

    # 패혈증 suppress_reduce
    from .symptom import SYMPTOM_DATA as SD
    for s in st.symptoms:
        if s.name == "패혈증" and s.suppressed == 0:
            turns = max(1, turns + SD["패혈증"].get("suppress_reduce", 0))

    chosen.suppressed = turns
    chosen.neglect = 0
    chosen.escalate_count = 0
    st.msg(f"    {chosen.name} {turns}턴 억제")
    trigger_suppress(st, chosen.name)

def _reveal(st: BattleState, n: int):
    hidden = [s for s in st.symptoms if True]  # 모든 증상은 알려진 것으로 처리
    st.msg(f"    증상 {n}개 의도 확인")

File: engine/card.js
const readline = require('readline');
const { triggerSuppress, getSymptomsData } = require('./symptom');

let CARD_DATA = null;

function loadCards(path) {
  const { loadJSON } = require('./state');
  const arr = loadJSON(path);
  CARD_DATA = {};
  for (const c of arr) CARD_DATA[c.id] = c;
}

function getCardData() { return CARD_DATA; }

async function playCard(st, cardId, rl, autoSuppressTarget = null) {
  const c = CARD_DATA[cardId];
  const e = c.effect;
  const SD = getSymptomsData();

  st.msg(`  > ${cardId} 사용`);

  if (c.type === 'treatment') {
    let dmg = e.damage || 0;
    // 반감 (고열경련)
    for (const s of st.symptoms) {
      if (s.suppressed === 0 && SD?.[s.name]?.treatment_half) {
        dmg = Math.floor(dmg / 2);
        st.msg(`    [${s.name}] 치료 효과 반감`);
        break;
      }
    }
    dmg = Math.max(0, dmg + st.treatmentBuff - st.treatmentDebuff);
    st.treatmentBuff = 0;

    if (e.suppress) {
      await doSuppress(st, e, rl, autoSuppressTarget);
    } else if (dmg) {
      st.diseaseHp -= dmg;
      st.msg(`    본체 -${dmg} HP → ${st.diseaseHp}`);
    }

  } else if (c.type === 'stabilize') {
    if (e.defense) {
      st.defenseTotal += e.defense;
      st.msg(`    방어 +${e.defense} (누적 ${st.defenseTotal})`);
    }
    if (e.heal) {
      st.patientHp = Math.min(st.maxPatientHp, st.patientHp + e.heal);
      st.msg(`    환자 +${e.heal} HP → ${st.patientHp}`);
    }
    if (e.reveal) st.msg(`    증상 ${e.reveal}개 의도 확인`);
    if (e.reveal_all) st.msg(`    모든 증상 의도 확인`);

  } else if (c.type === 'support') {
    if (e.buff_next_treatment) {
      st.treatmentBuff += e.buff_next_treatment;
      st.msg(`    다음 치료 카드 +${e.buff_next_treatment}`);
    }
    if (e.cost_reduce_next) {
      st.costReduceNext += e.cost_reduce_next;
      st.msg(`    다음 카드 코스트 -${e.cost_reduce_next}`);
    }
    if (e.draw) {
      st.drawCards(e.draw);
      st.msg(`    ${e.draw}장 드로우`);
    }
    if (e.buff_treatment_this_turn) {
      st.treatmentBuff += e.buff_treatment_this_turn;
      st.msg(`    이번 턴 치료 카드 +${e.buff_treatment_this_turn}`);
    }
    if (e.reveal_all) st.msg(`    모든 증상 의도 확인`);
  }
}

async function doSuppress(st, effect, rl, autoTarget = null) {
  const SD = getSymptomsData();
  const turns_base = effect.turns || 2;
  const active = st.symptoms.filter(s => s.suppressed === 0);
  if (!active.length) { st.msg('    억제할 활성 증상 없음'); return; }

  let chosen;
  if (effect.suppress === 'any') {
    if (autoTarget) {
      chosen = active.find(s => s.name === autoTarget);
    } else {
      console.log('    억제할 증상 선택:');
      active.forEach((s, i) => console.log(`      ${i}. ${s.name}`));
      const ans = await question(rl, '      번호: ');
      chosen = active[parseInt(ans)];
    }
    if (!chosen) { st.msg('    잘못된 입력'); return; }
  } else {
    chosen = active.find(s => s.name === effect.suppress);
    if (!chosen) { st.msg(`    ${effect.suppress} 활성 아님`); return; }
  }

  // 패혈증 suppress_reduce
  let turns = turns_base;
  for (const s of st.symptoms) {
    if (s.name === '패혈증' && s.suppressed === 0) {
      turns = Math.max(1, turns + (SD['패혈증']?.suppress_reduce || 0));
    }
  }

  chosen.suppressed = turns;
  chosen.neglect = 0;
  chosen.escalateCount = 0;
  st.msg(`    ${chosen.name} ${turns}턴 억제`);
  triggerSuppress(st, chosen.name);
}

function question(rl, prompt) {
  return new Promise(resolve => rl.question(prompt, resolve));
}

module.exports = { loadCards, playCard, getCardData, doSuppress };
